SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/DormancyTests/DynamicActorAwakeAfterDormantChangePropertyTest.cpp

```diff
@@ -17,33 +29,35 @@ void ADynamicActorAwakeAfterDormantChangePropertyTest::PrepareTest()
 
 	// Step 1 - Spawn dormancy actor and change NetDormancy to DORM_DormantAll
 	AddStep(TEXT("ServerSpawnDormancyActor"), FWorkerDefinition::Server(1), nullptr, [this]() {
-		AActor* Actor = CreateDormancyTestActor();
-		Actor->SetNetDormancy(DORM_DormantAll);
-		RegisterAutoDestroyActor(Actor);
+		DormancyActor = SpawnActor<ADormancyTestActor>(FActorSpawnParameters(), /*bRegisterAsAutoDestroy*/ true);
+		// Going from DORM_DormInitial to DORM_DormantAll
+		DormancyActor->SetNetDormancy(DORM_DormantAll);
 		FinishStep();
 	});
 
-	// Step 2 - Client check NetDormancy is DORM_DormantAll
-	AddStep(
-		TEXT("ClientRequireDormancyTestState"), FWorkerDefinition::AllClients, nullptr, nullptr,
-		[this](float DeltaTime) {
-			RequireDormancyActorCount(1);
-			RequireDormancyTestState(DORM_DormantAll, /*TestRepProperty*/ 0, /*ActorCount*/ 1);
-			FinishStep();
-		},
-		5.0f);
+	// Step 2 Require channel to actually be dormant
+	AddStep(TEXT("ServerAssertActuallyDormant"), FWorkerDefinition::Server(1), nullptr, nullptr, [this](float) {
+		const TSet<TSharedPtr<FNetworkObjectInfo>, FNetworkObjectKeyFuncs>& Actives = DormancyActor->GetNetDriver()->GetNetworkObjectList().GetActiveObjects();
+		// Require channel to actually be dormant
+		RequireEqual_Bool(Actives.Contains(DormancyActor), false, TEXT("Require test actor to no longer be in ActiveObjects"));
+		FinishStep();
+	});
+
+	// Step 3 - Client check NetDormancy is DORM_DormantAll
+	AddStep(("ClientAssertDormancyTestState"), FWorkerDefinition::AllClients, nullptr,[this](){
+		AssertEqual_Int(DormancyActor->NetDormancy, DORM_DormantAll, TEXT("Dormancy on ADormancyTestActor"));
+		AssertEqual_Int(DormancyActor->TestIntProp, 0, TEXT("TestIntProp on ADormancyTestActor"));
+		FinishStep();
+	});
 
-	// Step 3 - Server set TestIntProp to 1
+	// Step 4 - Server set TestIntProp to 1
 	AddStep(TEXT("ServerModifyNetDormancy"), FWorkerDefinition::Server(1), nullptr, [this]() {
-		for (TActorIterator<ADormancyTestActor> Iter(GetWorld()); Iter; ++Iter)
-		{
-			ADormancyTestActor* DormancyTestActor = *Iter;
-			DormancyTestActor->TestIntProp = 1;
-		}
+		DormancyActor->TestIntProp = 1;
 		FinishStep();
 	});
 
-	// Step 4 - Give chance for property to be replicated to clients
+	// Step 5 - Give chance for property to be replicated to clients, if we don't do this the
```

Comment:
Driveby - but if we want to really ensure it's replicated we should add a DormancyActor->ForceNetUpdate() after setting it to 1 in Step 4

Reply:
Oh - but we don't want it to actually replicated - right. Ignore me.